DiffusionPreprocessing/scripts/eddy_postproc.sh

```diff
@@ -95,42 +111,45 @@ ${FSLDIR}/bin/fslmaths ${datadir}/data -abs -Tmin -bin -fillh ${datadir}/fov_mas
 	 
 if [ ! $GdCoeffs = "NONE" ] ; then
     echo "Correcting for gradient nonlinearities"
-	# Note: "data_warped" is eddy-current and suspectibility distortion corrected (via 'eddy'), but prior to gradient distortion correction
-	# i.e., "data_posteddy_preGDC" would be another way to think of it
-    ${FSLDIR}/bin/immv ${datadir}/data ${datadir}/data_warped
+	  # Note: data in the warped directory is eddy-current and suspectibility distortion corrected (via 'eddy'), but prior to gradient distortion correction
+	  # i.e., "data_posteddy_preGDC" would be another way to think of it
+    mkdir -p ${datadir}/warped
+    ${FSLDIR}/bin/immv ${datadir}/data ${datadir}/warped/data_warped
+    ${FSLDIR}/bin/immv ${datadir}/fov_mask ${datadir}/warped/fov_mask_warped
+    ${FSLDIR}/bin/immv ${datadir}/cnr_maps ${datadir}/warped/cnr_maps_warped
 
     # Dilation outside of the field of view to minimise the effect of the hard field of view edge on the interpolation
-	DiffRes=`${FSLDIR}/bin/fslval ${datadir}/data_warped pixdim1`
+	  DiffRes=`${FSLDIR}/bin/fslval ${datadir}/warped/data_warped pixdim1`
     DilateDistance=`echo "$DiffRes * 4" | bc`  # Extrapolates the diffusion data up to 4 voxels outside of the FOV
-    ${CARET7DIR}/wb_command -volume-dilate ${datadir}/data_warped.nii.gz $DilateDistance NEAREST ${datadir}/data_warped_dilated.nii.gz
+    ${CARET7DIR}/wb_command -volume-dilate ${datadir}/warped/data_warped.nii.gz $DilateDistance NEAREST ${datadir}/warped/data_dilated.nii.gz
 
     # apply gradient distortion correction
-    ${globalscriptsdir}/GradientDistortionUnwarp.sh --workingdir="${datadir}" --coeffs="${GdCoeffs}" --in="${datadir}/data_warped_dilated" --out="${datadir}/data" --owarp="${datadir}/fullWarp"
+    ${globalscriptsdir}/GradientDistortionUnwarp.sh --workingdir="${datadir}" --coeffs="${GdCoeffs}" --in="${datadir}/warped/data_dilated" --out="${datadir}/data" --owarp="${datadir}/fullWarp"
+    ${FSLDIR}/bin/immv ${datadir}/fullWarp ${datadir}/warped
+    ${FSLDIR}/bin/immv ${datadir}/fullWarp_abs ${datadir}/warped
+    ${FSLDIR}/bin/imrm ${datadir}/warped/data_dilated
+
+    # Transform CNR maps
+    ${CARET7DIR}/wb_command -volume-dilate ${datadir}/warped/cnr_maps_warped.nii.gz $DilateDistance NEAREST ${datadir}/warped/cnr_maps_dilated.nii.gz
+    ${FSLDIR}/bin/applywarp --rel --interp=trilinear -i ${datadir}/warped/cnr_maps_dilated -r ${datadir}/warped/cnr_maps_dilated -w ${datadir}/warped/fullWarp -o ${datadir}/cnr_maps
```

Comment:
Did you intentionally choose trilinear rather than spline interpolation here?

Reply:
No, that should have been spline interpolation